course_discovery/apps/course_metadata/search_indexes/documents/common.py

```diff
@@ -201,9 +201,15 @@ class BaseCourseDocument(OrganizationsMixin, BaseDocument):
     image_url = fields.TextField()
     logo_image_urls = fields.TextField(multi=True)
     level_type = fields.TextField(analyzer=html_strip, fields={'raw': fields.KeywordField()})
-    partner = fields.TextField(analyzer=html_strip, fields={'raw': fields.KeywordField()})
+    partner = fields.TextField(
+        analyzer=html_strip,
+        fields={'raw': fields.KeywordField(), 'lower': fields.TextField(analyzer=case_insensitive_keyword)}
```

Comment:
Does using `TextField` for `lower` mean we are doing the full/slow search thing? Can we use an analyzer on a KeywordField? I'm guessing not.

Reply:
Search should be neither slow nor full-text. I use a special tokenizer for this https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-keyword-tokenizer.html 
I have read a lot of documentation and chose the solution that is often used in such cases. I believe that speed should not drop significantly.

> Can we use an analyzer on a KeywordField?

I remember that even on edx-notes-api I wanted to use analyzers for a KeywordField, but I couldn't do it due to the very nature of this field.

Reply:
:+1: